### PR TITLE
MDEV-30879 Add conversion to based 62 for CONV function

### PR DESCRIPTION
**Pull request by Weijun-H**
- [x] *The Jira issue number for this PR is: MDEV-30879*

## Description
This patch adds conversion to based 62 for CONV function.

This patch splits the CONV function into two situation: first, convert the number from 2 to 36, second, convert the number from 36 to 62. Use `convert_str_to_ull` and `convert_str_to_ll` to handle the case lager than 36.
## How can this PR be tested?
Append the tests to `func_str`

## Basing the PR against the correct MariaDB version
- [x] *This is a new feature and the PR is based against the latest MariaDB development branch*
- [ ] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced*

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.

### CODE REVIEW

File: sql/item_strfunc.cc [hunk 1]
```diff
@@ -3712,6 +3712,202 @@ String *Item_func_lpad::val_str(String *str)
...
+  ulonglong cutoff;
+  uint cutlim;
+  ulonglong i;
+  const char *s, *e;
+  const char *save;
+  int overflow;
+
+  *err= 0;				/* Initialize error indicator */
+
+  s = nptr;
```
Comment (LinuxJedi) [inline sql/item_strfunc.cc:3733]: No space before the equals here and a few other places in the code.

**Review by LinuxJedi — CHANGES_REQUESTED** [review]
Many thanks for your contribution. A few minor changes needed I think. Please also add some tests for the extremes such as overflows and maximum allowed / minimum allowed for negatives.

**Comment by LinuxJedi** [comment]
Judging by other implementations and this wiki page (https://en.wikipedia.org/wiki/Base62), the case processing is inverted in this patch. I'm working on a fix for it.

**Comment by LinuxJedi** [comment]
I've re-written this in #2589 so that it is in the charset code instead of duplicates of the charset code. Many thanks for your contribution, this guided the final patch and some of your tests were used (as marked by the Co-authored-by tag).
